### PR TITLE
Azure-IoT-Sdk for C release for 08-30-2021

### PR DESCRIPTION
**Pull request by momuno**
**Describe the pull request**
This updates the prior PR #19644
Attempt to fix how c-utility CMakeLists.txt finds CURL so that vcpkg can build.

### CODE REVIEW

**Comment by momuno** [comment]
Troubleshooting new build errors.

**Comment by momuno** [comment]
Hi @JonLiu1993, I have been working through these errors and am blocked.  Here is an example error log:

```
CMake Error at /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:574 (_add_library):
  Target "umqtt" links to target "OpenSSL::SSL" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:145 (add_library)
```

However, this `add_library` is not supposed to link to any target.

```
add_library(umqtt ${source_c_files} ${source_h_files})
```

The linking to the IMPORTED target `aziotsharedutil` is supposed to occur a few lines later:
```
target_link_libraries(umqtt aziotsharedutil)
```

To double check the `aziotsharedutil` had linked the "missing" targets, the `get_target_property` call for `INTERFACE_LINK_LIBRARIES` on the `aziotsharedutil` returns the following: (this is local to my machine)
```
$<$<NOT:$<CONFIG:DEBUG>>:/mnt/c/b/vcpkg-1/installed/x64-linux/lib/libcurl.a>$<$<CONFIG:DEBUG>:/mnt/c/b/vcpkg-1/installed/x64-linux/debug/lib/libcurl-d.a>dl-lpthreadOpenSSL::SSLOpenSSL::CryptoZLIB::ZLIB/mnt/c/b/vcpkg-1/installed/x64-linux/lib/libssl.a/mnt/c/b/vcpkg-1/installed/x64-linux/lib/libcrypto.a-lpthreaddldl-lpthreadpthreadmrt
```

I looked for the `_add_library` defintion to investigate further, but haven't found it
I am at out of ideas at the moment.  Can you please take a look to see what I might be missing?
Thanks.


NOTE: 
`find_package(azure_c_shared_utility)` is used, but the imported TARGET appears to be `aziotsharedutil`.  
The `azure-c-shared-utilityConfig.cmake` file contains the following:

```
include("${CMAKE_CURRENT_LIST_DIR}/azure_c_shared_utilityTargets.cmake")

get_target_property(AZURE_C_SHARED_UTILITY_INCLUDES aziotsharedutil INTERFACE_INCLUDE_DIRECTORIES)

set(AZURE_C_SHARED_UTILITY_INCLUDES ${AZURE_C_SHARED_UTILITY_INCLUDES} CACHE INTERNAL "")

include("${CMAKE_CURRENT_LIST_DIR}/azure_c_shared_utilityFunctions.cmake")
include("${CMAKE_CURRENT_LIST_DIR}/azure_iot_build_rules.cmake")
```

**Comment by momuno** [comment]
@JonLiu1993 `qt5-activeqt:x64-windows` is a new regression.  This issue was not around 3 days ago when the tests ran.

**Comment by JackBoosY** [comment]
```
PS G:\19809\vcpkg-1> .\vcpkg.exe depend-info azure-iot-sdk-c:x64-linux
azure-macro-utils-c:
openssl:
vcpkg-cmake:
vcpkg-cmake-config:
zlib:
curl[non-http, openssl, ssl]: openssl, vcpkg-cmake, vcpkg-cmake-config, zlib
umock-c: azure-macro-utils-c
azure-c-shared-utility: azure-macro-utils-c, curl, openssl, umock-c
azure-uamqp-c: azure-c-shared-utility, azure-macro-utils-c, umock-c
azure-uhttp-c: azure-c-shared-utility, azure-macro-utils-c, umock-c
azure-umqtt-c: azure-c-shared-utility, azure-macro-utils-c, umock-c
parson:
azure-iot-sdk-c: azure-c-shared-utility, azure-macro-utils-c, azure-uamqp-c, azure-uhttp-c, azure-umqtt-c, parson, umock-c
```

`azure-iot-sdk-c` depends on `azure-c-shared-utility`, and `azure-c-shared-utility` depends on `cur[openssl]` which requires openssl.

**Comment by JackBoosY** [comment]
The reason why x64-linux failed is because `azure-c-shared-utility` depends on `curl` but didn't contains `find_depedency(CURL CONFIG)` on its `azure_c_shared_utilityConfig.cmake`.

**Comment by momuno** [comment]
@JackBoosY thank you for your help on this. I didn't realize find_dependency(CURL CONFIG) needs to be in the azure_c_shraed_utilityConfig.cmake since the CMakeLists.txt already contains find_package(CURL CONFIG).  Looking at
https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html, I think its making more sense to me now.

**Comment by JackBoosY** [comment]
@momuno  See https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html

When we use `find_package (CURL)`, `FindCURL.cmake` will provide the absolute path of the curl library and header files, and write them into `<Tt>argets.cmake`.
However, in vcpkg, due to the existence of `vcpkg-cmake-wrapper`, `find_package(CURL CONFIG)` is automatically called when `find_package(CURL)` is used. At this time, what is written in `<Tt>argets.cmake` is the CURL target, **NOT** the absolute path. So at this time you need to add `find_dependency(CURL CONFIG)` in `<Cc>onfig.cmake`.

File: ports/azure-c-shared-utility/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          ]
+        },
+        {
+          "name": "curl",
+          "platform": "linux"
+        },
+        {
+          "name": "openssl",
+          "platform": "linux"
+        },
```
Comment (dg0yt) [inline ports/azure-c-shared-utility/vcpkg.json:36]: IIUC curl and openssl are core dependencies and don't need to be repeated in the features.

**Comment by dg0yt** [comment]
> When we use `find_package (CURL)`, `FindCURL.cmake` will provide the absolute path of the curl library and header files, and write them into `<Tt>argets.cmake`.
> However, in vcpkg, due to the existence of `vcpkg-cmake-wrapper`, `find_package(CURL CONFIG)` is automatically called when `find_package(CURL)` is used. At this time, what is written in `<Tt>argets.cmake` is the CURL target, **NOT** the absolute path. So at this time you need to add `find_dependency(CURL CONFIG)` in `<Cc>onfig.cmake`.

I don't think this is accurate. In particular for CURL, CMake will now always use the config if it is available, at least implicitly. And the config provides all variables that the find module used to provide. 
But *it is entirely up to the consuming project* if it uses target `CURL::libcurl` or variable `CURL_LIBRARIES` in order to refer to CURL. Modern CMake prefers targets.

>  I didn't realize find_dependency(CURL CONFIG) needs to be in the azure_c_shared_utilityConfig.cmake since the azure_c_shared_utility CMakeLists.txt already contains find_package(CURL). 

The key point is: 
`CMakeLists.txt` finds curl when *building* azure_c_shared_utility. 
`azure_c_shared_utilityConfig.cmake` finds curl when *using* azure_c_shared_utility. 

In particular for static linkage, you need not only a single curl library ("absolute path"), but a number of additional libraries to be linked to the consuming binaries. Of course you might hard code these dependencies at build time, but this isn't good practice. It enough when each package (config) knows its direct dependencies, and defers lookup of transitive dependencies.

**Comment by JackBoosY** [comment]
https://github.com/microsoft/vcpkg/blob/71422c627264daedcbcd46f01f1ed0dcd8460f1b/ports/curl/vcpkg-cmake-wrapper.cmake#L7
The cmake wrapper **does** change the default find behavior, which will also affect the use in `<Cc>onfig.cmake`.
If a port uses vcpkg's internal dependencies, the find behavior when building it will always be changed by the cmake wrapper.
Of course, using the Find module inside the port is always the highest priority, but in order to adapt to vcpkg, we do not recommend this.

**Comment by dg0yt** [comment]
> The cmake wrapper **does** change the default find behavior, which will also affect the use in `<Cc>onfig.cmake`.

I didn't want to dispute that. I just wanted to highlight that: 
- CMake's [FindCURL.cmake](https://cmake.org/cmake/help/latest/module/FindCURL.html) uses the config files, too:
  > If CURL was built using the CMake buildsystem then it provides its own CURLConfig.cmake file for use with the find_package() command's config mode. This module looks for this file and, if found, returns its results with no further action.
- Independent of find behaviour, `find_package (CURL)` just provides targets and variables.
  > When we use `find_package (CURL)`, `FindCURL.cmake` will provide the absolute path of the curl library and header files, and write them into `<Tt>argets.cmake`. 
  > ...
  > At this time, what is written in `<Tt>argets.cmake` is the CURL target, NOT the absolute path.

  `find_package (CURL)` / `FindCURL.cmake` won't write anything. What is written depends on the implementation in azure_c_shared_utility, not on vcpkg.

**Comment by JackBoosY** [comment]
> `find_package (CURL)` / `FindCURL.cmake` won't write anything. What is written depends on the implementation in azure_c_shared_utility, not on vcpkg.

It does affect. There are 2 scenes:
1. Use `find_package(CURL)`:
`FindCURL.cmake` will provides the macros `CURL_INCLUDE_DIRS` / `CURL_LIBRARIES` which contains the absolue path, then cmake will set them to `INTERFACE_INCLUDE_DIRECTORIES` and `INTERFACE_LINK_LIBRARIES`.

2. Use `find_package(CURL CONFIG)`:
`CURLConfig.cmake` will provides the target `CURL::curl` and cmake will set them to `INTERFACE_INCLUDE_DIRECTORIES` and `INTERFACE_LINK_LIBRARIES`.

That's what I said: different search methods will affect the use process.

**Comment by dg0yt** [comment]
> There are 2 scenes:
> 
> 1. Use `find_package(CURL)`:
>    `FindCURL.cmake` will provides the macros `CURL_INCLUDE_DIRS` / `CURL_LIBRARIES` which contains the absolue path, then cmake will set them to `INTERFACE_INCLUDE_DIRECTORIES` and `INTERFACE_LINK_LIBRARIES`.
> 2. Use `find_package(CURL CONFIG)`:
>    `CURLConfig.cmake` will provides the target `CURL::curl` and cmake will set them to `INTERFACE_INCLUDE_DIRECTORIES` and `INTERFACE_LINK_LIBRARIES`.
> 
> That's what I said: different search methods will affect the use process.

I'm sorry, but I don't agree. This seems to confuse lookup with usage. AFAICT both forms of lookup (with or without `CONFIG`) are equivalent, setting up variables and a target. So both forms allow two variants of usage:

1. `target_link_libraries(foo ${CURL_LIBRARIES})` will only affect linking (legacy CMake) (unless `CURL_LIBRARIES` contains the target name instead of absolute paths). Extra commands are needed for include paths, definitions, etc.
2. `target_link_libraries(foo CURL::libcurl)` will affect linking, include paths, definition etc., via target properties (Modern CMake).

**Comment by momuno** [comment]
> However, in vcpkg, due to the existence of `vcpkg-cmake-wrapper`, `find_package(CURL CONFIG)` is automatically called when `find_package(CURL)` is used. At this time, what is written in `<Tt>argets.cmake` is the CURL target, **NOT** the absolute path. So at this time you need to add `find_dependency(CURL CONFIG)` in `<Cc>onfig.cmake`.

If `find_dependency()` wraps `find_package()` (see https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html), and the `vcpkg-cmake-wrapper` causes `find_package(CURL)` to call `find_package(CURL CONFIG)`, then wouldn't `find_dependency(CURL)` suffice?

**Comment by momuno** [comment]
Hello @dg0yt and @JackBoosY, thank you for your discourse on the topic.

To test, I updated the azure-c-shared-utility `CMakeLists.txt` to contain`find_package(CURL)` and its `azure-c-shared-utilityConfig.cmake` to contain `find_dependency(CURL CONFIG)`.  The problem has been resolved for x64-linux.

However, now the x86-windows build fails.  Below are my tests and log results.

1.
- `CMakeLists.txt` contains `find_package(CURL)`. 
- `azure-c-shared-utilityConfig.cmake` contains `find_dependency(CURL CONFIG)`.  

OR

- `CMakeLists.txt` contains `find_package(CURL CONFIG)`. 
- `azure-c-shared-utilityConfig.cmake` contains `find_dependency(CURL CONFIG)`. 
 
Log error:
```
CMake Error at C:/a/vcpkg_e5825374-d775-4ae4-8251-0c9140996378/scripts/buildsystems/vcpkg.cmake:784 (_find_package):
  Could not find a package configuration file provided by "CURL" with any of
  the following names:

    CURLConfig.cmake
    curl-config.cmake

  Add the installation prefix of "CURL" to CMAKE_PREFIX_PATH or set
  "CURL_DIR" to a directory containing one of the above files.  If "CURL"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.21/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  C:/a/vcpkg_e5825374-d775-4ae4-8251-0c9140996378/installed/x86-windows/share/azure_c_shared_utility/azure_c_shared_utilityConfig.cmake:5 (find_dependency)
  C:/a/vcpkg_e5825374-d775-4ae4-8251-0c9140996378/scripts/buildsystems/vcpkg.cmake:784 (_find_package)
  CMakeLists.txt:60 (find_package)
```

Per the instructions in the log, I see a `curl-config.cmake.in` file in the curl repo under `curl/CMake`, but no `curl-config.cmake`.   I attempted to add `set(CURL_DIR CMake)` before both `find_package(CURL CONFIG)` and `find_dependency(CURL CONFIG)`, but the error remains.  Am I messing up the syntax?


2.
- `CMakeLists.txt` contains `find_package(CURL)`. 
- `azure-c-shared-utilityConfig.cmake` contains `find_dependency(CURL)`.  

OR

- `CMakeLists.txt` contains `find_package(CURL)`. 
- `azure-c-shared-utilityConfig.cmake` contains `find_dependency(CURL)`.  
and move Targets.cmake before the find_dependency call:
```
include("${CMAKE_CURRENT_LIST_DIR}/azure_c_shared_utilityTargets.cmake")

include(CMakeFindDependencyMacro)
find_dependency(CURL)
```

Log error:
```
CMake Error at C:/Program Files/CMake/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR)
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.21/Modules/FindCURL.cmake:181 (find_package_handle_standard_args)
  C:/a/vcpkg_e5825374-d775-4ae4-8251-0c9140996378/scripts/buildsystems/vcpkg.cmake:784 (_find_package)
  C:/Program Files/CMake/share/cmake-3.21/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  C:/a/vcpkg_e5825374-d775-4ae4-8251-0c9140996378/installed/x86-windows/share/azure_c_shared_utility/azure_c_shared_utilityConfig.cmake:5 (find_dependency)
  C:/a/vcpkg_e5825374-d775-4ae4-8251-0c9140996378/scripts/buildsystems/vcpkg.cmake:784 (_find_package)
  CMakeLists.txt:60 (find_package)
```

**Comment by JackBoosY** [comment]
Please note that only suitable for vcpkg since curl in vcpkg provides `CURLConfig.cmake` and vcpkg toolchain will call it automaticly.

**Comment by dg0yt** [comment]
- You want consistent results, so please use `CONFIG` consistently with `find_package` and `find_dependency`. The keyword excludes Find modules.
- The reported Call Stack doesn't list `share/curl/vcpkg-cmake-wrapper.cmake`. This is strange. Can you verify that you don't have installed an outdated version of port curl:x86-windows? (`vcpkg remove --recurse curl:x86-windows`)

**Comment by dg0yt** [comment]
Maybe I better comment in https://github.com/Azure/azure-c-shared-utility/pull/555.

**Comment by dg0yt** [comment]
Okay, finally some insight. `azure_c_shared_utility` uses `${CURL_LIBRARIES}` for linking to curl. This is legacy CMake, but not wrong. Howerver, we have issues with `CURL_LIBRARIES` in vpckg:
- The curl wrapper doesn't properly carry curl's interface link libraries to `CURL_LIBRARIES`. In fact, it gets the list, and throws it away...  
  https://github.com/microsoft/vcpkg/blame/master/ports/curl/vcpkg-cmake-wrapper.cmake#L13-L27
  This must be fixed in the wrapper.
- The interface link libraries list contains not only resolved libraries, but also targets. 
  The wrapper must ensure that the targets are defined.

From my local `packages/curl_x64-linux/share/curl/CURLTargets.cmake`:
~~~cmake
set_target_properties(CURL::libcurl PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "CURL_STATICLIB"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "dl;-lpthread;OpenSSL::SSL;OpenSSL::Crypto;ZLIB::ZLIB"
)
~~~

**Comment by dg0yt** [comment]
Okay, scrap the last comment.

For Windows, azure_c_shared_utility doesn't use - and doesn't depend on - port curl. That's why `find_dependency(CURL)` doesn't work on Windows, unless it is pulled in another port - but you don't need `find_dependency(CURL)` on Windows. 
(Using `CURL_LIBRARIES`, you shouldn't need `find_dependency(CURL)` at all in the exported config file.)

**Comment by momuno** [comment]
> * The reported Call Stack doesn't list `share/curl/vcpkg-cmake-wrapper.cmake`. This is strange. Can you verify that you don't have installed an outdated version of port curl:x86-windows? (`vcpkg remove --recurse curl:x86-windows`)

On x86_windows, curl was not installed at all via vcpkg.  Should it have been?
When I run `vcpkg install azure-iot-sdk-c`, it does not install.  

```
PS C:\a\vcpkg_e5825374-d775-4ae4-8251-0c9140996378> .\vcpkg.exe install azure-iot-sdk-c
Computing installation plan...
The following packages will be built and installed:
  * azure-c-shared-utility[core]:x86-windows -> 2021-09-07
    azure-iot-sdk-c[core]:x86-windows -> 2021-09-07
  * azure-macro-utils-c[core]:x86-windows -> 2020-06-17#2
  * azure-uamqp-c[core]:x86-windows -> 2020-09-07
  * azure-uhttp-c[core]:x86-windows -> 2021-09-07
  * azure-umqtt-c[core]:x86-windows -> 2021-09-07
  * parson[core]:x86-windows -> 2020-09-14
  * umock-c[core]:x86-windows -> 2020-06-17-1
  ```

**Comment by momuno** [comment]
> For Windows, azure_c_shared_utility doesn't use - and doesn't depend on - port curl.

This I think explains why I don't see it installed.

File: ports/azure-c-shared-utility/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "azure-c-shared-utility",
+  "version-string": "2021-09-08",
```
Comment (dg0yt) [inline ports/azure-c-shared-utility/vcpkg.json:3]: Can this be turned to `version-date`?
Cf. https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md#version-fields
(Similar for the related ports.)

Comment (momuno) [reply ports/azure-c-shared-utility/vcpkg.json:3]: We would like to use the name of the release in the future, which could be a date or "LTS..."  version-string seems the best option for that.

Comment (BillyONeal) [reply ports/azure-c-shared-utility/vcpkg.json:3]: @momuno The consequence of that though is that your versions are unorderable. If version-string reflects what your actual policy is, then we should leave it that way, but I think your customers would prefer if your versions were orderable.

Comment (momuno) [reply ports/azure-c-shared-utility/vcpkg.json:3]: I see. Ok that works too.

**Comment by JackBoosY** [comment]
When building azure-iot-sdk:x64-osx:
```
[44/98] /Library/Developer/CommandLineTools/usr/bin/cc -DARCHITECTURE_x86_64=1 -DMACOSX -I/Users/vagrant/Data/installed/x64-osx/share/azure_macro_utils_c/include -I/Users/vagrant/Data/installed/x64-osx/share/inc -I/Users/vagrant/Data/installed/x64-osx/share/pal/linux -I/Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/iothub_client/../deps/parson -I/Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/provisioning_client/inc -I/Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/provisioning_client/inc/internal -I/Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/iothub_client/inc -isystem /Users/vagrant/Data/installed/x64-osx/include -isystem /Users/vagrant/Data/installed/x64-osx/include/azureiot -fPIC -Werror -DUSE_EDGE_MODULES -DUSE_PROV_MODULE -g -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk -fPIC -std=gnu99 -MD -MT iothub_client/CMakeFiles/iothub_client_amqp_ws_transport.dir/src/iothubtransport_amqp_telemetry_messenger.c.o -MF iothub_client/CMakeFiles/iothub_client_amqp_ws_transport.dir/src/iothubtransport_amqp_telemetry_messenger.c.o.d -o iothub_client/CMakeFiles/iothub_client_amqp_ws_transport.dir/src/iothubtransport_amqp_telemetry_messenger.c.o -c /Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/iothub_client/src/iothubtransport_amqp_telemetry_messenger.c
FAILED: iothub_client/CMakeFiles/iothub_client_amqp_ws_transport.dir/src/iothubtransport_amqp_telemetry_messenger.c.o 
/Library/Developer/CommandLineTools/usr/bin/cc -DARCHITECTURE_x86_64=1 -DMACOSX -I/Users/vagrant/Data/installed/x64-osx/share/azure_macro_utils_c/include -I/Users/vagrant/Data/installed/x64-osx/share/inc -I/Users/vagrant/Data/installed/x64-osx/share/pal/linux -I/Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/iothub_client/../deps/parson -I/Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/provisioning_client/inc -I/Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/provisioning_client/inc/internal -I/Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/iothub_client/inc -isystem /Users/vagrant/Data/installed/x64-osx/include -isystem /Users/vagrant/Data/installed/x64-osx/include/azureiot -fPIC -Werror -DUSE_EDGE_MODULES -DUSE_PROV_MODULE -g -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk -fPIC -std=gnu99 -MD -MT iothub_client/CMakeFiles/iothub_client_amqp_ws_transport.dir/src/iothubtransport_amqp_telemetry_messenger.c.o -MF iothub_client/CMakeFiles/iothub_client_amqp_ws_transport.dir/src/iothubtransport_amqp_telemetry_messenger.c.o.d -o iothub_client/CMakeFiles/iothub_client_amqp_ws_transport.dir/src/iothubtransport_amqp_telemetry_messenger.c.o -c /Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/iothub_client/src/iothubtransport_amqp_telemetry_messenger.c
/Users/vagrant/Data/buildtrees/azure-iot-sdk-c/src/608a2338ea-063215add1.clean/iothub_client/src/iothubtransport_amqp_telemetry_messenger.c:1027:80: error: cast to smaller integer type 'TELEMETRY_MESSENGER_EVENT_SEND_COMPLETE_RESULT' (aka 'enum TELEMETRY_MESSENGER_EVENT_SEND_COMPLETE_RESULT_TAG') from 'const void *' [-Werror,-Wvoid-pointer-to-enum-cast]
        TELEMETRY_MESSENGER_EVENT_SEND_COMPLETE_RESULT messenger_send_result = (TELEMETRY_MESSENGER_EVENT_SEND_COMPLETE_RESULT)action_context;
                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by momuno** [comment]
Ready to be merged.

**Review by JackBoosY — APPROVED** [review]

**Comment by momuno** [comment]
@JackBoosY Thanks for your approval.  Will someone with write access merge this PR, or am I supposed to have write access (I don't currently).

**Comment by BillyONeal** [comment]
Thanks for the updates!

**Pull request merged**